### PR TITLE
:gear: rlm_python: Fix handling of attribute lists

### PR DESCRIPTION
**Pull request by jpereira**
It fixes the way of how the .py scripts receives the 'dict' within these
major lists. e.g: `p["request"]`, `p["control"]`, `p["reply"]` and `p["session-state"]`

Also, allowing the capability to update them returning a tuple with such a dictionary.

### CODE REVIEW

File: src/modules/rlm_python/rlm_python.c [hunk 1]
```diff
@@ -488,95 +488,147 @@ static int mod_populate_vptuple(rlm_python_t const *inst, request_t *request, Py
...
+	 * If some list is not available, NONE is used instead
+	 */
+	p_arg = PyTuple_New(4);
+	if (!p_arg) {
+		ERROR("%s - Memory cannot be allocated for PyTyple_New", funcname);
+		rcode = RLM_MODULE_FAIL;
+		goto finish;
+	}
+
+	/* If there is a request, fill in the first 4 attribute lists */
```
Comment (alandekok) [inline src/modules/rlm_python/rlm_python.c:568]: there should always be a request?  If the function is being called from `recv foo` or xlat, there's always a request.

File: src/modules/rlm_python/rlm_python.c [hunk 2]
```diff
@@ -488,95 +488,147 @@ static int mod_populate_vptuple(rlm_python_t const *inst, request_t *request, Py
...
+			rcode = RLM_MODULE_FAIL;
+			goto finish;
 		}
 	}
 
 	/* Call Python function. */
-	p_ret = PyObject_CallFunctionObjArgs(p_func, p_arg, NULL);
+	p_dict = PyDict_New();
+	if (!p_dict ||
+	    PyDict_SetItemString(p_dict, "request", PyTuple_GET_ITEM(p_arg, 0)) ||
```
Comment (alandekok) [inline src/modules/rlm_python/rlm_python.c:590]: Just move this name setting to `mod_populate_vps()`, and pass the name to that function.  less code == better code

File: src/modules/rlm_python/rlm_python.c [hunk 3]
```diff
@@ -587,19 +639,33 @@ static unlang_action_t do_python_single(rlm_rcode_t *p_result,
...
+
+		/* process updateDict */
+		update_dict = PyTuple_GET_ITEM(p_ret, 1);
+		if (!PyDict_CheckExact(update_dict)) {
+			ERROR("%s - updateDict is not dictionary", funcname);
+			rcode = RLM_MODULE_FAIL;
+			goto finish;
+		}
+
+		mod_vptuple(request->packet, inst, request, &request->request_pairs,
```
Comment (alandekok) [inline src/modules/rlm_python/rlm_python.c:654]: TBH macros here could shorten this code.  Repetition is opportunity for mistakes

File: src/tests/modules/python/auth_with_shared_storage.unlang [hunk 4]
```diff
@@ -1,13 +1,13 @@
 # Should return noop (no shared variable set)
-pmod_shared_storage
+pmod7_shared_storage
```
Comment (alandekok) [inline src/tests/modules/python/auth_with_shared_storage.unlang:2]: Why this change?  I don't understand...

Comment (jpereira) [reply src/tests/modules/python/auth_with_shared_storage.unlang:2]: Just to keep the same as the previous tests with numbers. e.g: `pmod${number}_$name`
